File: 10_min_waiting_time_V1.py
# O(nlogn) Time
# O(1) Space

def minimumWaitingTime(queries):
    totalWaitingTime = 0
    remainQueries = len(queries)
    
    queries.sort()
    
    for i in range(len(queries)):

        #  后面每个queries都要等这么长时间, 还剩几个query就等几倍
        remainQueries = len(queries) - i - 1
        totalWaitingTime += remainQueries * queries[i]

    return totalWaitingTime

queries = [3, 1, 2, 2, 6]
print(minimumWaitingTime(queries))

File: 10_min_waiting_time_V1.cpp
#include <vector>
#include <string>
#include <iostream>

using namespace std;

int minimumWaitingTime(vector<int> queries) {
    // O(nlogn) Time
    // O(1) Space
    int sum = 0;
    int waitTime = 0;

    sort(queries.begin(), queries.end());

    for(int i = 0; i < queries.size() - 1; ++i) {
        cout << "It's: " << queries[i] << "   Before sum: " << sum;

        waitTime = waitTime + queries[i];
        sum += waitTime;

        cout << "   After sum: " << sum << endl;
    } // for

    return sum;
} // minimumWaitingTime


int main() {
    vector<int> queries = {3, 2, 1, 2, 6};

    cout << minimumWaitingTime(queries);
}